datadog_checks_dev/datadog_checks/dev/tooling/commands/validate/readmes.py

```diff
@@ -0,0 +1,78 @@
+# (C) Datadog, Inc. 2020-present
+# All rights reserved
+# Licensed under a 3-clause BSD style license (see LICENSE)
+import re
+from os import path
+
+import click
+
+from ...utils import complete_valid_checks, get_root, get_valid_integrations, read_readme_file
+from ..console import CONTEXT_SETTINGS, abort, echo_failure, echo_info, echo_success
+
+IMAGE_EXTENSIONS = {".png", ".jpg"}
+
+
+@click.command(context_settings=CONTEXT_SETTINGS, short_help='Validate README.md files')
+@click.pass_context
+@click.argument('integration', autocompletion=complete_valid_checks, required=False)
+def readmes(ctx, integration):
+    """Validates README files
+
+    If `check` is specified, only the check will be validated,
+    otherwise all README files in the repo will be.
+    """
+
+    repo = ctx.obj['repo_name']
+
+    errors = False
+    integrations = []
+
+    if integration:
+        integrations = [integration]
+    else:
+        integrations = sorted(get_valid_integrations())
+
+    for integration in integrations:
+        has_overview = False
+        has_setup = False
+
+        lines = read_readme_file(integration)
+        for line_no, line in lines:
+
+            if "## Overview" == line.strip():
+                has_overview = True
+
+            if "## Setup" == line.strip():
+                has_setup = True
+
+            for ext in IMAGE_EXTENSIONS:
+                if ext in line:
+                    IMAGE_REGEX = (
+                        rf".*https:\/\/raw\.githubusercontent\.com\/DataDog\/"
+                        rf"{re.escape(repo)}\/master\/({re.escape(integration)}\/images\/.*.{ext}).*"
+                    )
+
+                    match = re.match(IMAGE_REGEX, line)
+                    if not match:
+                        errors = True
+                        echo_failure(f"{integration} readme file does not have a valid image file on line {line_no}")
+                        echo_info(
+                            f"This image path must be in the form: "
+                            f"https://raw.githubusercontent.com/DataDog/{repo}/master/{integration}/images/<IMAGE_NAME>"
+                        )
+
+                    rel_path = match.groups()[0]
```

Comment:
I think need to abort above or put this in an else, otherwise we never see the failure message above if match doesn't exist.

Reply:
EDIT: sorry you do see the error message above, but it gets drowned out a bit by a stacktrace

Reply:
Good catch, I updated to break if there's a failure!